lib/src/component/abstract_transition.dart

```diff
@@ -184,7 +187,14 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps, S
       skipCount = 0;
     }
 
+    var timer = new Timer(transitionTimeout, () {
+      assert(ValidationUtil.warn(
+        'The number of transitions expected to complete have not completed. Something is most likely wrong.'
+      ));
```

Comment:
Something else that I was thinking about. Should we still call `complete` if the timer completes? I could go either way, was just wondering what the desired behavior is.